envoy/config/filter/http/jwt_authn/v2/config.proto

```diff
@@ -0,0 +1,197 @@
+  // It is the cluster name in the Envoy "cluster_manager" config section.
+  // In order for Envoy to call "jwks_uri", its host has to be specified
+  // as a "cluster" in the config for each jwks_uri.
+  string jwks_uri_cluster = 4;
+
+  // Duration after which the cached public key should be expired. The
+  // system wide default is applied if no duration is explicitly
+  // specified.
+  google.protobuf.Duration public_key_cache_duration = 5;
+
+  // If false, the JWT is removed in the request after a success verification.
+  // Set it true, if don't want it to be removed.
+  google.protobuf.BoolValue forward_jwt = 6;
+
+  // If true, the request is allowed if the JWT verification fails
+  // Default is not allowed.
+  google.protobuf.BoolValue allow_failed_jwt = 7;
+
+  // Two fields below define where to extract the JWT from an HTTP request.
+  //
+  // If no explicit location is specified, the following default locations are
+  // tried in order:
+  //
+  // 1. The Authorization header using the Bearer schema.
+  //    See `here <https://tools.ietf.org/html/rfc6750#section-2.1>`_
+  //    Example::
+  //
+  //       Authorization: Bearer <token>.
+  //
+  // 2. `access_token` query parameter.
+  //    See `here <https://tools.ietf.org/html/rfc6750#section-2.3>`_
+  //
+  //
+  // Only the first JWT token is verified per request if multiple tokens are
+  // specified.
+  //
+  // If both jwt_headers and jwt_params are specified, jwt_headers will be
+  // checked first.
+  //
+
+  // Specify the HTTP headers to extract JWT token.
+  //
+  // For examples, following config::
+  //
+  //   jwt_headers:
+  //   -name: x-goog-iap-jwt-assertion
+  //
+  // can be used to extract token from header::
+  //
+  //   x-goog-iap-jwt-assertion: <JWT>.
+  //
+  repeated JwtHeader jwt_headers = 8;
+
+  // JWT is sent in a query parameter. `jwt_params` represents the
+  // query parameter names.
+  //
+  // For example, if config is::
+  //
+  //   jwt_params:
+  //     jwt_token
+  //
+  // The query parameter is::
+  //
+  //    /path?jwt_token=<JWT>
+  //
+  repeated string jwt_params = 9;
+}
+
+// This message specifies a header location to extract JWT token.
+// A HTTP header format is::
+//
+//    name: value_prefix<JWT>
+//
+message JwtHeader {
+  // The HTTP header name.
+  string name = 1;
+
+  // The value prefix. The value format is "prefix<token>"
+  // For example, for "Authorization: Bearer <token>",
+  // value_prefix="Bearer " with a space at the end.
+  string value_prefix = 2;
+}
+
+// This message defines a pattern to match a HTTP request.
+// A pattern is matched only if all "headers" are matched.
+message HttpMatcher {
+  // Specifies a set of headers should match on.
+  // For example::
+  //
+  //    headers:
+  //    - name: :method
+  //      value: OPTIONS
+  //    headers:
+  //    - name: :path
+  //      regex_match: /.*
+  //
+  repeated envoy.api.v2.route.HeaderMatcher headers = 1;
```

Comment:
Can you comment (in this PR review) on why this was preferred over `RouteMatch`?

Reply:
Initially I felt that RouteMatch is too complicated,  especially its "runtime" field, it may not apply here. 

Well, I look it over again.   I changed my mind,  we could just use it directly.  Will update